jgrapht-ext/src/main/java/org/jgrapht/ext/GraphMLExporter.java

```diff
@@ -223,29 +727,80 @@ public void export(Writer writer, Graph<V, E> g)
             handler.startElement("", "", "edge", attr);
 
             if (edgeLabelProvider != null) {
-                // <data>
-                attr.clear();
-                attr.addAttribute("", "", "key", "CDATA", "edge_label");
-                handler.startElement("", "", "data", attr);
-
-                // Content for <data>
                 String edgeLabel = edgeLabelProvider.getEdgeName(e);
-                handler.characters(
-                    edgeLabel.toCharArray(),
-                    0,
-                    edgeLabel.length());
-                handler.endElement("", "", "data");
+                if (edgeLabel != null) {
+                    writeData(handler, "edge_label_key", edgeLabel);
+                }
+            }
+
+            if (exportEdgeWeights) {
+                Double weight = g.getEdgeWeight(e);
+                if (weight != 1.0) { // not default value
+                    writeData(
+                        handler,
+                        "edge_weight_key",
+                        String.valueOf(weight));
+                }
+            }
+
+            // find edge attributes
+            Map<String, String> edgeAttributes = null;
+            if (edgeAttributeProvider != null) {
+                edgeAttributes = edgeAttributeProvider
+                    .getComponentAttributes(e);
+            }
+            if (edgeAttributes == null) {
+                edgeAttributes = Collections.emptyMap();
+            }
+
+            // check all registered
+            for (Entry<String, AttributeDetails> entry : registeredAttributes
+                .entrySet())
+            {
+                AttributeDetails details = entry.getValue();
+                if (details.category.equals(AttributeCategory.EDGE)
+                    || details.category.equals(AttributeCategory.ALL))
+                {
+                    String name = entry.getKey();
+                    String defaultValue = details.defaultValue;
+                    if (edgeAttributes.containsKey(name)) {
+                        String value = edgeAttributes.get(name);
+                        if (defaultValue == null
+                            || !value.equals(defaultValue))
+                        {
+                            if (value != null) {
```

Comment:
this if clause should never fail, e.g. value can never be null, otherwise value.equals in the encapsulating clause would have thrown a nullpointer